### ISSUE TITLE
Add interface to suppress "Rerouting..." expression in NavigationView 

### ISSUE THREAD

**Issue by OttyLab** [mapbox/mapbox-navigation-android#3726]
**Android API: No dependency**
**Mapbox Navigation SDK version: v1.1.2**

### Steps to trigger behavior
 1. Call `MapboxNavigation.setRerouteController` with `null` to disable rerouting
 2. Go out side of route
 
### Expected behavior
"Rerouting..." in `InstructionView` and `ProgressBar` in `SummaryBottomSheet` are not shown.

### Actual behavior
"Rerouting..." in `InstructionView` and `ProgressBar` in `SummaryBottomSheet` are shown.
![feature_request](https://user-images.githubusercontent.com/13183117/97872130-da327700-1d58-11eb-9059-f55ceeb41b1c.gif)

### Detailed information
For current implementation, this is an expected behaviour  ([InstructionView](https://github.com/mapbox/mapbox-navigation-android/blob/v1.1.2/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java#L235)/[SummaryBottomSheet](https://github.com/mapbox/mapbox-navigation-android/blob/v1.1.2/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/summary/SummaryBottomSheet.java#L121)).

At the same time, showing those information is a bit confusing because it's not rerouting actually. Instead, if SDK have an interface to suppress those behaviour or to customize the expression (e.g. show "Off route"), it would be more user-friendly.

**Comment by shoaibalisabir** [comment]
public boolean allowRerouteFrom(Point offRoutePoint) {
        return false;
    }

**Comment by P-Zenker** [comment]
> public boolean allowRerouteFrom(Point offRoutePoint) {
>     return false;
> }

Yeah, this was for old versions. The current behaviour is very confusing.

**Comment by abhishek1508** [comment]
Thanks for using the Mapbox Navigation SDK for Android and being a valued customer.

Mapbox will be soon deprecating any support for `v0` and `v1` versions of the SDK. To facilitate this transition we’re launching a new drop-in UI component into `v2`, equivalent to the existing `NavigationView` `v1` in its design goals, however with a more modern and customizable API. 

We plan to launch this new drop-in UI component as a Developer Preview feature in April, as part of the `v2.5` series. Since you are using `NavigationView` with `v1`, we’d love to hear your feedback so that we can incorporate it ahead of a GA release.

If you’re interested in having early access to the upcoming drop-in UI for `v2` and its documentation, drop a comment on this ticket or send an email to abhishek.kejriwal@mapbox.com

/cc @zugaldia @AhmerKhan1
